src/autiter.rs

```diff
@@ -53,13 +53,13 @@ pub trait Automaton<P>: Sized {
     }
 
     /// Returns an iterator of non-overlapping matches in `s`.
-    fn find<'a, 's>(
+    fn find<'a, 's, Q: ?Sized + AsRef<[u8]>>(
         &'a self,
-        s: &'s str,
+        s: &'s Q,
     ) -> Matches<'a, 's, P, Self> {
```

Comment:
Take a look at how the `Iterator` trait does it: https://github.com/rust-lang/rust/blob/master/src/libcore/iter.rs#L489

Can you apply those same techniques here? What happens?